cesm2_albedo/calculate_sw_alb.jl

```diff
@@ -0,0 +1,98 @@
+using NCDatasets
+
+# Load the data
+rsds_file = NCDataset("rsds_Amon_CESM2_historical_r1i1p1f1_gn_185001-201412.nc")
+rsus_file = NCDataset("rsus_Amon_CESM2_historical_r1i1p1f1_gn_185001-201412.nc")
+
+# Extract the data
+rsds = rsds_file["rsds"][:, :, :]
+rsus = rsus_file["rsus"][:, :, :]
+
+# Create output file
+sw_alb_file = NCDataset("sw_albedo_Amon_CESM2_historical_r1i1p1f1_gn_185001-201412_v2_no-nans.nc", "c")
+
+# Define the dimensions
+defDim(sw_alb_file, "lat", rsus_file.dim["lat"])
+defDim(sw_alb_file, "bnds", rsus_file.dim["nbnd"])
+defDim(sw_alb_file, "lon", rsus_file.dim["lon"])
+defDim(sw_alb_file, "time", rsus_file.dim["time"])
+
+# copy file attributes
+for (attrib_name,attrib_value) in rsus_file.attrib
+    sw_alb_file.attrib[attrib_name] = attrib_value
+end
+
+# for each var in the dataset, copy the needed attributes and set the others
+sw_attrib = []
+for (attrib_name,attrib_value) in rsus_file["rsus"].attrib
+    if attrib_name != "coordinates"
+        push!(sw_attrib, attrib_name => attrib_value)
+    end
+end
+
+time_bnds_attrib = []
+no_copy_time_bnds = ["calendar", "units"]
+for (attrib_name,attrib_value) in rsus_file["time_bnds"].attrib
+    if !(attrib_name in no_copy_time_bnds)
+        push!(time_bnds_attrib, attrib_name => attrib_value)
+    end
+end
+
+time_attrib = []
+no_copy_time = ["calendar", "title", "type"]
+for (attrib_name,attrib_value) in rsus_file["time"].attrib
+    if !(attrib_name in no_copy_time)
+        push!(time_attrib, attrib_name => attrib_value)
+    end
+end
+push!(time_attrib, "calendar" => "365_day")
+
+lon_bnds_attrib = []
+for (attrib_name,attrib_value) in rsus_file["lon_bnds"].attrib
+    if attrib_name != "units"
+        push!(lon_bnds_attrib, attrib_name => attrib_value)
+    end
+end
+
+lat_bnds_attrib = []
+for (attrib_name,attrib_value) in rsus_file["lat_bnds"].attrib
+    if attrib_name != "units"
+        push!(lat_bnds_attrib, attrib_name => attrib_value)
+    end
+end
+
+lon_lat_attrib_no_copy = ["title", "type", "valid_max", "valid_min"]
+lon_attrib = []
+for (attrib_name,attrib_value) in rsus_file["lon"].attrib
+    if !(attrib_name in lon_lat_attrib_no_copy)
+        push!(lon_attrib, attrib_name => attrib_value)
+    end
+end
+push!(lon_attrib, "long_name" => "longitude")
+
+lat_attrib = []
+for (attrib_name,attrib_value) in rsus_file["lat"].attrib
+    if !(attrib_name in lon_lat_attrib_no_copy)
+        push!(lat_attrib, attrib_name => attrib_value)
+    end
+end
+push!(lat_attrib, "long_name" => "latitude")
+
+# Calculate the shortwave albedo
+sw_alb = ((x, y) -> y != 0.0 ? x/y : 1.0f0).(rsus, rsds)
+
+# create the variables
+defVar(sw_alb_file, "lat", rsus_file["lat"],  dimnames(rsus_file["lat"]); attrib = lat_attrib)
+lat_bnds = defVar(sw_alb_file, "lat_bnds", Float64, ("bnds", "lat"); attrib = lat_bnds_attrib)
```

Comment:
could you also set this with  `defVar(sw_alb_file, "lat_bnds", rsus_file["lat_bnds"], dimnames(rsus_file["lat_bnds"]; attrib = lat_bnds_attrib)`? it would be good to define all the variables with as similar calls as possible, but if it's not possible that's okay

Reply:
I made this change, but this makes `time_bnds` have additional attributes not present in the original file. This is why the `lat_bnds` and `lon_bnds` also were set differently, but I got their attributes to match the original file. 

<img width="1015" alt="Screenshot 2024-10-07 at 2 35 33 PM" src="https://github.com/user-attachments/assets/52cb0304-28fc-45c0-bda6-ec6fdf6381f3">